xgcm/grid.py

```diff
@@ -1721,6 +1658,15 @@ def _1d_grid_ufunc_dispatch(
         if isinstance(axis, str):
             axis = [axis]
 
+        # This function is restricted to a single data input, so we need to check the input validity
+        # here early.
+        # TODO: This will fail if a sequence of inputs is passed, but not with a very helpful error
+        # TODO: message. @TOM do you think it is worth to check the type and raise another error in that case?
```

Comment:
@TomNicholas should I add an additional check that explicitly raises a more helpful error if multiple arguments are passed?